app/uk/gov/hmrc/catalogue/github/GithubConfig.scala

```diff
@@ -38,42 +39,40 @@ object GithubConfig extends GithubConfig {
   val githubHiddenRepositoriesConfigKey = "github.hidden.repositories"
   val githubHiddenTeamsConfigKey = "github.hidden.teams"
 
-  def githubOpen = fallBackToFileSystem(".credentials", GithubCredentials(
-    config(s"$githubOpenConfigKey.host"),
-    config(s"$githubOpenConfigKey.user"),
-    config(s"$githubOpenConfigKey.key")))
+  private val gitOpenConfig = (key: String) => config(s"$githubOpenConfigKey.$key")
+  private val gitEnterpriseConfig = (key: String) => config(s"$githubEnterpriseConfigKey.$key")
+
+
+  lazy val githubOpenCredentials = credentials(gitOpenConfig).fold(fromFileSystem(".credentials"))(identity)
 
-  def githubEnterprise = fallBackToFileSystem(".githubenterprise", GithubCredentials(
-    config(s"$githubEnterpriseConfigKey.host"),
-    config(s"$githubEnterpriseConfigKey.user"),
-    config(s"$githubEnterpriseConfigKey.key")))
+  lazy val githubEnterpriseCredentials = credentials(gitEnterpriseConfig).fold(fromFileSystem(".githubenterprise"))(identity)
 
-  def hiddenRepositories = config(githubHiddenRepositoriesConfigKey).split(",").toList
+  lazy val hiddenRepositories = config(githubHiddenRepositoriesConfigKey).fold(List.empty[String])(x => x.split(",").toList)
```

Comment:
its just a matter of choice I find fold as much natural choice to operate over Options